src/truncate.jl

```diff
@@ -73,6 +73,8 @@ Truncated(d::UnivariateDistribution, l::Integer, u::Integer) = Truncated(d, floa
 
 params(d::Truncated) = tuple(params(d.untruncated)..., d.lower, d.upper)
 partype(d::Truncated) = partype(d.untruncated)
+Base.eltype(::Type{Truncated{D, S, T} } ) where {D, S, T <: Real} = T
```

Comment:
```suggestion
Base.eltype(::Type{Truncated{D, S, T} } ) where {D, S} = T
```
The type constraint is not necessary

Reply:
ok